src/host/directio.cpp

```diff
@@ -994,14 +994,14 @@ void EventsToUnicode(_Inout_ std::deque<std::unique_ptr<IInputEvent>>& inEvents,
         {
             // We find the offset into the original buffer by the dimensions of the original request rectangle.
             ptrdiff_t rowOffset = 0;
-            RETURN_IF_FAILED(PtrdiffTSub(target.Y, requestRectangle.Top(), &rowOffset));
-            RETURN_IF_FAILED(PtrdiffTMult(rowOffset, requestRectangle.Width(), &rowOffset));
+            RETURN_HR_IF(E_ABORT, !base::CheckSub<LONGLONG>(target.Y, requestRectangle.Top()).AssignIfValid(&rowOffset));
+            RETURN_HR_IF(E_ABORT, !base::CheckMul<LONGLONG>(rowOffset, requestRectangle.Width()).AssignIfValid(&rowOffset));

```

Comment:
Shouldn't all of these PtrdiffT calculations be using `ptrdiff_t` for the type? I don't think it's the same as `LONGLONG` in a 32bit build.